backend/tests/test_api.py

```diff
@@ -2,3 +2,62 @@ def test_hello(client):
     response = client.get("/")
     assert response.status_code == 200
     assert response.json() == {"message": "Hello user! Tip: open /docs or /redoc for documentation"}
+
+
+def test_ping(client):
+    response = client.get("/ping")
+    assert response.status_code == 200
+    assert response.json().get("ping").startswith("pong @")
+
+
+def test_import_from_github(client):
+    response = client.post(
+        "/test/testing_branch/import",
+        json={"description": "test_description"},
+    )
+    assert response.status_code == 200
+    assert response.json() is True
+
+
+def test_upload_taxonomy(client):
+    with open("tests/data/test.txt", "rb") as f:
+        response = client.post(
+            "/test_taxonomy/test_branch/upload",
+            files={"file": f},
+            data={"description": "test_description"},
+        )
+    assert response.status_code == 200
+    assert response.json() is True
+
+
+def test_add_taxonomy_duplicate_branch_name(client):
+    with open("tests/data/test.txt", "rb") as f:
+        response = client.post(
+            "/test_taxonomy_2/test_branch/upload",
+            files={"file": f},
+            data={"description": "test_description"},
+        )
+    assert response.status_code == 409
+    assert response.json() == {"detail": "branch_name: Branch name should be unique!"}
+
+
+def test_add_taxonomy_invalid_branch_name(client):
+    with open("tests/data/test.txt", "rb") as f:
+        response = client.post(
+            "/test_taxonomy/invalid-branch-name/upload",
+            files={"file": f},
+            data={"description": "test_description"},
+        )
+    assert response.status_code == 400
+    assert response.json() == {"detail": "branch_name: Enter a valid branch name!"}
+
+
+def test_add_taxonomy_duplicate_project_name(client):
```

Comment:
same as for test above, your test must not depend on the last state of another test.